tests/test_mask.py

```diff
@@ -225,15 +284,46 @@ def make_outfile(mask_sites, in_file, out_file, cleanup=True):
         mask.run(args)
         assert os.path.exists(out_file), "Output file incorrectly deleted"
 
-    def test_run_normal_case(self, bed_file, vcf_file, tmpdir, argparser, mp_context):
-        test_outfile = str(tmpdir / "out")
-        def check_args(mask_sites, in_file, out_file, cleanup):
+    def test_run_normal_case(self, bed_file, vcf_file, out_file, argparser, mp_context):
+        def check_args(mask_sites, in_file, _out_file, cleanup):
             assert mask_sites == TEST_BED_SEQUENCE, "Wrong mask sites provided"
             assert in_file == vcf_file, "Incorrect input file provided"
-            assert out_file == test_outfile, "Incorrect output file provided"
+            assert _out_file == out_file, "Incorrect output file provided"
             assert cleanup is True, "Cleanup erroneously passed in as False"
-            open(out_file, "w").close() # want to test we don't delete output.
         mp_context.setattr(mask, "mask_vcf", check_args)
-        args = argparser("--mask=%s --sequences=%s --output=%s" %(bed_file, vcf_file, test_outfile))
+        args = argparser("--mask=%s --sequences=%s --output=%s" %(bed_file, vcf_file, out_file))
+        mask.run(args)
+        assert os.path.exists(out_file), "Output file incorrectly deleted"
+    
+    def test_run_with_mask_sites(self, vcf_file, out_file, argparser, mp_context):
+        args = argparser("--mask-sites 2 8 -s %s -o %s" % (vcf_file, out_file))
+        def check_mask_sites(mask_sites, *args, **kwargs):
+            assert mask_sites == [2,8]
+        mp_context.setattr(mask, "mask_vcf", check_mask_sites)
+        mask.run(args)
```

Comment:
This test would be more helpful if it ran `mask_vcf` and confirmed that the requested sites were no longer present in the output file. I actually love the way you've setup the argparser fixture here such that we can essentially run end-to-end tests from the unit testing framework. I know it is a little more computationally expensive, but running the whole program this way can catch more errors.

We also need a corresponding test for masking specific sites with a FASTA input. This test will fail right now because mask_sites values are not converted to 0-based coordinates (and because of the mask from beginning/end None issue mentioned earlier).

```
def test_run_with_mask_sites_for_fasta(self, fasta_file, out_file, argparser):
    # 1-based coordinates from the user should translate to 0-based masking of the FASTA sequence.
    args = argparser("--mask-sites 1 -s %s -o %s" % (fasta_file, out_file))
    mask.run(args)
    output = SeqIO.parse(out_file, "fasta")
    for record in output:
        # Site 1 in the sequence is the 0-indexed site in the sequence record.
        assert record.seq[0] == "N"
```

Reply:
That is so cool! Thank you!